src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -429,6 +429,24 @@ private static long ParseTimeValue(JToken jToken, string claimName)
             throw LogHelper.LogExceptionMessage(new FormatException(LogHelper.FormatInvariant(LogMessages.IDX14300, LogHelper.MarkAsNonPII(claimName), jToken.ToString(), LogHelper.MarkAsNonPII(typeof(long)))));
         }
 
+        internal static string SafeLogJwtToken(object obj)
+        {
+            if (obj == null)
+                return string.Empty;
+
+            // not a string
+            if (!(obj is string token))
+                return obj.ToString();
+
+            int lastDot = token.LastIndexOf(".");
+
+            // no dots, maybe not a JWT
+            if (lastDot == -1)
```

Comment:
>lastDot [](http://example.com/codeflow?start=16&length=7)

Since this is meant to parse JWT token strings we should throw an argument exception here.

Reply:
We will throw in the ReadToken() method if the string token is not an encoded JWT token.

Reply:
This should be safe regardless of the caller, for non strings, non tokens we should return null.